### PR TITLE
Fix multiline comments + significant-indentation processing

### LINKED ISSUE #2436: parse error since 4.4.25
### build.sbt

```scala
libraryDependencies += "org.scalameta" %% "scalameta" % "4.4.25"

scalaVersion := "2.13.6"
```

### Main.scala

```scala
import scala.meta._

object Main {

  def main(args: Array[String]): Unit = {

    val result = s"""
class A1 extends scala.annotation.StaticAnnotation
class A2 extends scala.annotation.StaticAnnotation

@A1
/*
 * hello
 */
@A2
class B
""".parse[Source].get

    println(result)
  }
}
```

### output

```
[info] running Main 
[error] (run-main-4) <input>:8: error: expected start of definition
[error]  */
[error]    ^
[error] <input>:8: error: expected start of definition
[error]  */
[error]    ^
[error] 	at scala.meta.internal.parsers.Reporter.syntaxError(Reporter.scala:16)
[error] 	at scala.meta.internal.parsers.Reporter.syntaxError$(Reporter.scala:16)
[error] 	at scala.meta.internal.parsers.Reporter$$anon$1.syntaxError(Reporter.scala:22)
[error] 	at scala.meta.internal.parsers.Reporter.syntaxError(Reporter.scala:17)
[error] 	at scala.meta.internal.parsers.Reporter.syntaxError$(Reporter.scala:17)
[error] 	at scala.meta.internal.parsers.Reporter$$anon$1.syntaxError(Reporter.scala:22)
[error] 	at scala.meta.internal.parsers.ScalametaParser.tmplDef(ScalametaParser.scala:4551)
[error] 	at scala.meta.internal.parsers.ScalametaParser.topLevelTmplDef(ScalametaParser.scala:4525)
[error] 	at scala.meta.internal.parsers.ScalametaParser$$anonfun$topStat$1.applyOrElse(ScalametaParser.scala:5064)
[error] 	at scala.meta.internal.parsers.ScalametaParser$$anonfun$topStat$1.applyOrElse(ScalametaParser.scala:5052)
[error] 	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:35)
[error] 	at scala.meta.internal.parsers.ScalametaParser.statSeq(ScalametaParser.scala:5042)
[error] 	at scala.meta.internal.parsers.ScalametaParser.topStatSeq(ScalametaParser.scala:5051)
[error] 	at scala.meta.internal.parsers.ScalametaParser.$anonfun$batchSource$1(ScalametaParser.scala:5282)
[error] 	at scala.meta.internal.parsers.ScalametaParser.atPos(ScalametaParser.scala:882)
[error] 	at scala.meta.internal.parsers.ScalametaParser.autoPos(ScalametaParser.scala:916)
[error] 	at scala.meta.internal.parsers.ScalametaParser.batchSource(ScalametaParser.scala:5241)
[error] 	at scala.meta.internal.parsers.ScalametaParser.$anonfun$source$1(ScalametaParser.scala:5224)
[error] 	at scala.meta.internal.parsers.ScalametaParser.atPos(ScalametaParser.scala:882)
[error] 	at scala.meta.internal.parsers.ScalametaParser.autoPos(ScalametaParser.scala:916)
[error] 	at scala.meta.internal.parsers.ScalametaParser.source(ScalametaParser.scala:5223)
[error] 	at scala.meta.internal.parsers.ScalametaParser.entrypointSource(ScalametaParser.scala:5229)
[error] 	at scala.meta.internal.parsers.ScalametaParser.$anonfun$parseSource$2(ScalametaParser.scala:143)
[error] 	at scala.meta.internal.parsers.ScalametaParser.parseRule(ScalametaParser.scala:53)
[error] 	at scala.meta.internal.parsers.ScalametaParser.parseSource(ScalametaParser.scala:143)
[error] 	at scala.meta.parsers.Parse$.$anonfun$parseSource$1(Parse.scala:29)
[error] 	at scala.meta.internal.parsers.ScalametaParser$$anon$264.apply(ScalametaParser.scala:5293)
[error] 	at scala.meta.parsers.Api$XtensionParseDialectInput.parse(Api.scala:25)
[error] 	at scala.meta.parsers.Api$XtensionParseInputLike.parse(Api.scala:14)
[error] 	at Main$.main(Main.scala:17)
[error] 	at Main.main(Main.scala)
```

### PR DESCRIPTION
**Pull request by dos65**
Fixes #2436

### CODE REVIEW

**Review by tgodzik — APPROVED** [review]
LGTM, I actually thought that something might be off last time, but the tests seemed to pass ok. We should check afterwards if it fixes the tests in Scalafmt, but I can do it on Monday.

File: scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala [hunk 1]
```diff
@@ -499,7 +499,10 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
           (nextRegions, currRef)
         } else if (curr.is[EOF]) {
           sepRegions match {
-            case x :: xs if x.isIndented => (xs, mkOutdent(currPos))
+            case x :: xs if x.isIndented =>
+              // do no include trailing comment/whitespace tokens
+              val pointPos = if (currPos - prevPos == 1) currPos else prevPos + 1
+              (xs, mkOutdent(pointPos))
```
Comment (dos65) [inline scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala:505]: This fixes what @kitbellew reported in [that comment](https://github.com/scalameta/scalameta/pull/2426#issuecomment-889733564)

**Comment by dos65** [comment]
@tgodzik I included an additional fix that finally solves problems that were spotted by scalafmt tests. I checked them locally - everything pass.

File: scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala [hunk 2]
```diff
@@ -499,7 +499,10 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
           (nextRegions, currRef)
         } else if (curr.is[EOF]) {
           sepRegions match {
-            case x :: xs if x.isIndented => (xs, mkOutdent(currPos))
+            case x :: xs if x.isIndented =>
+              // do no include trailing comment/whitespace tokens
+              val pointPos = if (currPos - prevPos == 1) currPos else prevPos + 1
```
Comment (kitbellew) [inline scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala:504]: looks like prevpos + 1 always

Comment (dos65) [reply scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala]: haha, thx!

**Review by kitbellew — APPROVED** [review]

**Pull request merged**